installinator/src/write.rs

```diff
@@ -759,21 +768,56 @@ impl ControlPlaneZoneWriteContext<'_> {
             .register();
     }
 
-    fn mupdate_override_artifact(
+    async fn mupdate_override_artifact(
         &self,
         mupdate_uuid: MupdateOverrideUuid,
-    ) -> BufList {
-        // Might be worth writing out individual hash IDs for each zone in the
-        // future.
+    ) -> Result<BufList, JoinError> {
         let hash_ids =
             [self.host_phase_2_id.clone(), self.control_plane_id.clone()]
                 .into_iter()
                 .collect();
-        let mupdate_override = MupdateOverrideInfo { mupdate_uuid, hash_ids };
+        let zones = compute_zone_hashes(&self.zones).await?;
+
+        let mupdate_override =
+            MupdateOverrideInfo { mupdate_uuid, hash_ids, zones };
         let json_bytes = serde_json::to_vec(&mupdate_override)
             .expect("this serialization is infallible");
-        BufList::from(json_bytes)
+        Ok(BufList::from(json_bytes))
+    }
+}
+
+/// Computes the zone hash IDs.
+///
+/// Hash computation is done in parallel on blocking tasks. If a task panics
+/// (should not happen in normal use), a `JoinError` is returned.
```

Comment:
Do we need to return an error at all? I think we've pretty liberally unwrapped `tokio::spawn()` await points where there's obviously no cancellation possible, because that means

(a) in prod we would have aborted due to the task panic anyway
(b) in dev we only propagate other panics

right? This seems pretty similar.

Reply:
I guess something I could do is to check whether the `JoinError` is a panic, and if so, panic as you suggested. If it's a cancellation then report that.

Reply:
Ehhh, what would cause the runtime to shutdown, in practice? We're running under a `#[tokio::main]`, right?

Reply:
Oh, if we have tests that exercise this and we might hit the whole "runtime cancels tasks in a random order when test ends" thing, then maybe we don't want to panic here. But I don't know that it's worth doing work to bubble any kind of error up to wicket for a case that's only possible in tests.

Reply:
Do you feel like it's a lot of work? I'm just generally concerned with dead air from a hard-to-introspect service I guess.

Reply:
No, it's not a lot of work. But handling this error in this way is pretty different from how we handle tokio `JoinError`s in general. E.g., even from within installinator: https://github.com/oxidecomputer/omicron/blob/03081e1bf4ef360be611148f6235087a555743e8/installinator-common/src/raw_disk_writer.rs#L86-L97

I don't think handling this error is _wrong_. I just think it's weird when we don't bother to most anywhere else, and it's unfortunate to make an otherwise-infallible function return a `Result`.

Reply:
All right, addressed.

Reply:
Thanks!